tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py

```diff
@@ -0,0 +1,241 @@
+):
+    """Simple EOF creation from a deployed EOF container"""
+    eof_state_test(
+        data=Container(
+            sections=[
+                eofcreate_code_section,
+                returncontract_sub_container,
+            ],
+        ),
+        container_post=Account(storage={slot_code_worked: value_code_worked}),
+    )
+
+
+def test_simple_create_from_creation(
+    eof_state_test: EOFStateTestFiller,
+):
+    """Simple EOF creation from a create transaction container"""
+    eof_state_test(
+        data=Container(
+            sections=[
+                returncontract_code_section,
+                Section.Container(container=Container(sections=[Section.Code(code=Op.STOP)])),
+            ],
+        ),
+        container_post=Account(storage={slot_code_worked: value_code_worked}),
+    )
+
+
+@pytest.mark.parametrize(
+    "zero_section",
+    [eofcreate_code_section, returncontract_code_section],
+    ids=["eofcreate", "returncontract"],
+)
+def test_reverting_container(
+    eof_state_test: EOFStateTestFiller,
+    zero_section: Container,
+):
+    """Test revert containers"""
+    eof_state_test(
+        data=Container(
+            sections=[
+                zero_section,
+                revert_sub_container,
+            ],
+        ),
+        container_post=Account(storage={slot_code_worked: value_code_worked}),
+    )
+
+
+@pytest.mark.parametrize(
+    "zero_sub_container,first_sub_container",
+    [
+        (eofcreate_code_section, returncontract_sub_container),
+        (returncontract_code_section, stop_sub_container),
+    ],
+    ids=["eofcreate", "returncontract"],
+)
+@pytest.mark.parametrize(
+    "extra_sub_container",
+    [stop_sub_container, revert_sub_container, returncontract_sub_container],
+    ids=["stop", "revert", "returncontract"],
+)
+def test_orphan_container(
+    eof_test: EOFTestFiller,
+    zero_sub_container: Container,
+    first_sub_container: Container,
+    extra_sub_container: Container,
+):
+    """Test orphaned containers"""
+    eof_test(
+        data=Container(
+            sections=[
+                zero_sub_container,
+                first_sub_container,
+                extra_sub_container,
+            ],
+        ),
+    )
+
+
+@pytest.mark.parametrize(
+    "zero_sub_container,first_sub_container",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            returncontract_sub_container,
+            id="EOFCREATE/RETURNCONTRACT",
+        ),
+        pytest.param(returncontract_code_section, stop_sub_container, id="RETURNCONTRACT/STOP"),
+        pytest.param(
+            returncontract_code_section, return_sub_container, id="RETURNCONTRACT/RETURN"
+        ),
+        pytest.param(eofcreate_code_section, revert_sub_container, id="EOFCREATE/REVERT"),
+        pytest.param(
+            returncontract_code_section, revert_sub_container, id="RETURNCONTRACT/REVERT"
+        ),
+    ],
+)
+def test_container_combos_valid(
```

Comment:
We should add invalid combos where one of many sections break the RETURN/STOP assumption for the init-container, as well as some valid combos where all sections comply with the assumptions with different combinations too.